code/modules/flufftext/Dreaming.dm

```diff
@@ -1,26 +1,22 @@
-/mob/living/carbon/proc/dream()
-	set waitfor = 0
-	dreaming = TRUE
-	var/list/dreams = list(
+GLOBAL_LIST_INIT(globalDreamMessages,list(
 		"an ID card","a bottle","a familiar face","a crewmember","a toolbox","a security officer","the captain",
 		"voices from all around","deep space","a doctor","the engine","a traitor","an ally","darkness",
 		"light","a scientist","a monkey","a catastrophe","a loved one","a gun","warmth","freezing","the sun",
 		"a hat","the Luna","a ruined station","a planet","plasma","air","the medical bay","the bridge","blinking lights",
 		"a blue light","an abandoned laboratory","Nanotrasen","The Syndicate","blood","healing","power","respect",
-		"riches","space","a crash","happiness","pride","a fall","water","flames","ice","melons","flying"
-		)
-	for(var/i in 1 to rand(1, rand(3, 7)))

```

Comment:
I dislike how you made dreams not happen as a sequence and instead happen randomly while sleeping